packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -199,14 +215,33 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
       port: 80
     };
 
+    if (props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS && props.certificate !== undefined) {
+      throw new Error('The HTTPS protocol must be used when a certificate is given');
+    }
+    const protocol = props.certificate ? ApplicationProtocol.HTTPS : (props.protocol ? props.protocol : ApplicationProtocol.HTTP);
```

Comment:
I would have expected the order to be:

```ts
props.protocol !== undefined ? props.protocol : (props.certificate ? ApplicationProtocol.HTTPS : ApplicationProtocol.HTTP)
```

?

That is, if a value is given it is honored (still subject to valiation), otherwise we pick the most appropriate default based on other parameters.

Reply:
Yeah thats better